Training/Model/NoteRepository.php

```diff
@@ -0,0 +1,229 @@
+     *
+     * @var CollectionFactory
+     */
+    private $noteCollectionFactory;
+
+    /**
+     * @var NoteSearchResultInterfaceFactory
+     */
+    private $searchResultFactory;
+
+    /**
+     * NoteRepository constructor.
+     *
+     * @param NoteInterface $note
+     * @param \Codifi\Training\Model\CustomerNoteFactory $noteFactory
+     * @param CustomerNoteResourse $noteResourse
+     * @param CollectionFactory $noteCollectionFactory
+     * @param NoteSearchResultInterfaceFactory $searchResultFactory
+     */
+    public function __construct(
+        NoteInterface $note,
+        CustomerNoteFactory $noteFactory,
+        CustomerNoteResourse $noteResourse,
+        CollectionFactory $noteCollectionFactory,
+        NoteSearchResultInterfaceFactory $searchResultFactory
+    ) {
+        $this->note = $note;
+        $this->noteFactory = $noteFactory;
+        $this->noteResourse = $noteResourse;
+        $this->noteCollectionFactory = $noteCollectionFactory;
+        $this->searchResultFactory = $searchResultFactory;
+    }
+
+    /**
+     * Get note by id.
+     *
+     * @param int $id
+     * @return object
+     * @throws NoSuchEntityException
+     */
+    public function getById($id): object
+    {
+        $noteModel = $this->noteFactory->create();
+        $this->noteResourse->load($noteModel, $id);
+        if (!$noteModel->getId()) {
+            throw new NoSuchEntityException(__('Unable to find hamburger with ID "%1"', $id));
+        }
+
+        return $noteModel;
+    }
+
+    /**
+     * Save note.
+     *
+     * @param NoteInterface $note
+     * @throws AlreadyExistsException
+     */
+    public function save(NoteInterface $note): void
+    {
+        $noteModel = $this->noteFactory->create();
+        $noteModel->setData($note);
+        $this->noteResourse->save($noteModel);
+    }
+
+    /**
+     * Delete note.
+     *
+     * @param NoteInterface $note
+     * @throws AlreadyExistsException
+     */
+    public function delete(NoteInterface $note): void
+    {
+        $noteModel = $this->noteFactory->create();
+        $this->noteResourse->delete($note);
+        $this->noteResourse->save($noteModel);
+    }
+
+    /**
+     * Delete note by id.
+     *
+     * @param int $noteId
+     * @throws NoSuchEntityException
+     */
+    public function deleteById($noteId) : void
+    {
+        $noteModel = $this->noteFactory->create();
+        $note = $this->noteResourse->load($noteModel, $noteId);
+        if (!$noteModel->getId()) {
+            throw new NoSuchEntityException(__('Unable to find hamburger with ID "%1"', $noteId));
+        }
+        $this->noteResourse->delete($note);
+        $this->noteResourse->save($noteModel);
+    }
+
+    /**
+     * Get list.
+     *
+     * @param SearchCriteriaInterface $searchCriteria
+     * @return mixed
+     */
+    public function getList(SearchCriteriaInterface $searchCriteria)
+    {
+        $collection = $this->noteCollectionFactory->create();
```

Comment:
Code can be simplified. See Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface and its using